### PR TITLE
change function return behaviour on error

### PR DESCRIPTION
**Pull request by rafaeljusto**
See #227

### CODE REVIEW

File: idn/punycode.go [hunk 1]
```diff
@@ -139,14 +143,16 @@ func tfunc(k, bias rune) rune {
 	return k - bias
 }
 
-// encode transforms Unicode input bytes (that represent DNS label) into punycode bytestream
+// encode transforms Unicode input bytes (that represent DNS label) into
+// punycode bytestream. This function would return nil if there's an invalid
+// character in the label
```
Comment (miekg) [inline idn/punycode.go]: label -> label.

**Comment by miekg** [comment]
@asergeyev think this looks ok (as did the other PR). What do you think?

File: idn/punycode.go (hunk 1 above)
Comment (rafaeljusto) [inline idn/punycode.go]: done!

**Comment by asergeyev** [comment]
Good to me now. Before it was just feeling irrational to return unicode, nothing was conceptually wrong with it.

**Pull request merged**